File: cases/taint/php/anonymous-implementation-positive/anonymous_implementation_positive.php
<?php
interface Handler
{
    public function handle(string $value): void;
}

function dfb_source(): string { // DFB-SOURCE: anonymous-implementation-input
    return "tainted";
}

function dfb_sink(string $value): void {} // DFB-SINK: anonymous-implementation-sink

function run(): void {
    $leak = new class implements Handler { // DFB-WITNESS: anonymous-implementation-bind
        public function handle(string $value): void
        {
            dfb_sink($value);
        }
    };
    $drop = new class implements Handler {
        public function handle(string $value): void
        {
            dfb_sink("clean");
        }
    };
    $leak->handle(dfb_source());
}

File: cases/taint/cpp/anonymous-implementation-positive/anonymous_implementation_positive.cpp
#include <functional>
#include <string>

std::string dfb_source() { // DFB-SOURCE: anonymous-implementation-input
    return "tainted";
}

void dfb_sink(const std::string &value) {} // DFB-SINK: anonymous-implementation-sink

void run() {
    std::function<void(const std::string &)> leaking =
        [](const std::string &value) { dfb_sink(value); }; // DFB-WITNESS: anonymous-implementation-instance
    std::function<void(const std::string &)> dropping =
        [](const std::string &value) { dfb_sink("clean"); };
    leaking(dfb_source());
}
